### ISSUE TITLE
XCode 14.3 archivation issue

### ISSUE THREAD

**Issue by mike-dutka** [CocoaPods/CocoaPods#11861]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

pod cache clean --all
pod update

XCode->Product-> Archive

## What did you expect to happen?

Install all pod dependencies correctly.
I can create a build archive for a project

## What happened instead?

Archiving fails with 

```
PhaseScriptExecution [CP]\ Embed\ Pods\ Frameworks ~/Library/Developer/Xcode/DerivedData/APP-cewpbmsjuchepdepqwhvevegsuju/Build/Intermediates.noindex/ArchiveIntermediates/APP/IntermediateBuildFilesPath/APP.build/Release-iphoneos/APP.build/Script-69CF97DCCC5B38497218EE47.sh (in target 'APP' from project 'APP')
    cd ~/Documents/Projects/
    /bin/sh -c ~/Library/Developer/Xcode/DerivedData/APP-cewpbmsjuchepdepqwhvevegsuju/Build/Intermediates.noindex/ArchiveIntermediates/APP/IntermediateBuildFilesPath/APP.build/Release-iphoneos/APP.build/Script-69CF97DCCC5B38497218EE47.sh

mkdir -p ~/Library/Developer/Xcode/DerivedData/APP-cewpbmsjuchepdepqwhvevegsuju/Build/Intermediates.noindex/ArchiveIntermediates/APP/BuildProductsPath/Release-iphoneos/APP.app/Frameworks
Symlinked...
rsync --delete -av --filter P .*.?????? --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "../../../IntermediateBuildFilesPath/UninstalledProducts/iphoneos/Alamofire.framework" "~/Library/Developer/Xcode/DerivedData/APP-cewpbmsjuchepdepqwhvevegsuju/Build/Intermediates.noindex/ArchiveIntermediates/APP/InstallationBuildProductsLocation/Applications/APP.app/Frameworks"
building file list ... rsync: link_stat "~/Documents/Projects/../../../IntermediateBuildFilesPath/UninstalledProducts/iphoneos/Alamofire.framework" failed: No such file or directory (2)
done

sent 29 bytes  received 20 bytes  98.00 bytes/sec
total size is 0  speedup is 0.00
rsync error: some files could not be transferred (code 23) at /AppleInternal/Library/BuildRoots/97f6331a-ba75-11ed-a4bc-863efbbaf80d/Library/Caches/com.apple.xbs/Sources/rsync/rsync/main.c(996) [sender=2.6.9]
Command PhaseScriptExecution failed with a nonzero exit code

```

The issue and the hotfix are described on an Apple Developer portal [here](https://developer.apple.com/forums/thread/725230) 

## CocoaPods Environment

```
CocoaPods : 1.12.0
Ruby : ruby 3.2.0 (2022-12-25 revision a528908271) [x86_64-darwin22]
RubyGems : 3.4.1
Host : macOS 13.3.1 (22E261)
Xcode : Version 14.3 (14E222b)
Git : git version 2.39.2 (Apple Git-143)
```

### Plugins

```
cocoapods-deintegrate : 1.0.5
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
```

**Comment by yberstad** [comment]
Having the same issue. This post claims that this should be fixed in 1.12.1, https://github.com/aws-amplify/amplify-flutter/issues/2818#issuecomment-1500706874. Do not know when the 1.12.1 will be realeased...

**Comment by bdaz** [comment]
Also running into this issue.

**Comment by fbronner** [comment]
Same here

**Comment by rodrigoliimaa** [comment]
Same problem here

**Comment by carloshpdoc** [comment]
Same problem here, folker!

**Comment by MortenHN81** [comment]
Same issue here, looking forward to .12.1 will be realeased

**Comment by MateuszLas421** [comment]
Maybe this will help  https://github.com/CocoaPods/CocoaPods/pull/11828

**Comment by TommyLeng** [comment]
I solved by following this.

https://github.com/CocoaPods/CocoaPods/issues/11808#issuecomment-1500907375

**Comment by TrabelsiAchraf** [comment]
Also running into this issue, looking forward to 1.12.1 will be released !

**Comment by iharandreyev** [comment]
[This solution](https://developer.apple.com/forums/thread/725230?answerId=746897022#746897022) did help.

**Comment by tim-eucalyptus** [comment]
Dear team, please release 1.12.1 for us.

**Comment by HsynKcmn** [comment]
Same problem I have.
Please release quick update for us.

**Comment by berkaykurkcu** [comment]
Seems like using XCode 14.2 and sticking to iOS 16.3 is the only way to prevent facing this issue for now. 
If you upgraded the iOS version, Apple has closed signing for 16.3

**Comment by rursache** [comment]
> Seems like using XCode 14.2 and sticking to iOS 16.3 is the only way to prevent facing this issue for now. If you upgraded the iOS version, Apple has closed signing for 16.3

or just move to the recommended way of doing things aka SPM

**Comment by paulb777** [comment]
1.12.1 released today

**Comment by mike-dutka** [comment]
1.12.1 has been released today. The issue has been resolved.

**Comment by geowarsong** [comment]
If someone still has an issue, you can add `post_install` hook in `podfile`

```
post_install do |installer|
  # Check if Xcode version is 14.3
  xcode_version = `xcodebuild -version | grep "Xcode" | awk '{print $2}'`.strip
  required_version = "14.3"

  if xcode_version == required_version
    puts "Checking and fixing Pods for Xcode 14.3 if needed..."

    file = 'Pods/Target Support Files/Pods-APPNAME/Pods-APPNAME-frameworks.sh'

    if File.exist?(file)
      content = File.read(file)

      if !content.include?('source="$(readlink -f "${source}")"')
        fixed_content = content.gsub('source="$(readlink "${source}")"', 'source="$(readlink -f "${source}")"')
        File.write(file, fixed_content)
        puts "Fixed file: #{file}"
      else
        puts "No fix needed for file: #{file}"
      end
    else
      puts "File not found: #{file}"
    end

  else
    puts "Xcode version is not 14.3, no action required."
  end
end
```

**Comment by KirillBarkousky** [comment]
I faced the same issue on Xcode 14.3 and also updated CocoaPods to version 1.12.1. 
However, our project doesn't have a 'Release' configuration, which is the reason for the issue. This occurred because the script didn't receive the PODS_ROOT variable, among others. I found a solution by deleting the 'Release' configuration from Pods.xcodeproj (located in the 'Info' tab of the project configuration view). The configuration files weren't set there.

**Comment by KirillBarkousky** [comment]
I faced the same issue on Xcode 14.3 and also updated CocoaPods to version 1.12.1.
However, our project doesn't have a 'Release' configuration, which is the reason for the issue. This occurred because the script didn't receive the PODS_ROOT variable, among others. I found a solution by deleting the 'Release' configuration from Pods.xcodeproj (located in the 'Info' tab of the project configuration view). The configuration files weren't set there.
